src/cdk/tree/padding.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directionality} from '@angular/cdk/bidi';
+import {Directive, ElementRef, Input, OnDestroy, Optional, Renderer2} from '@angular/core';
+import {takeUntil} from 'rxjs/operator/takeUntil';
+import {Subject} from 'rxjs/Subject';
+import {CdkTreeNode} from './node';
+import {CdkTree} from './tree';
+
+/**
+ * Indent for the children tree nodes.
+ * This directive will add left-padding to the node to show hierarchy.
+ */
+@Directive({
+  selector: '[cdkTreeNodePadding]',
+})
+export class CdkTreeNodePadding<T> implements OnDestroy {
+  /** Subject that emits when the component has been destroyed. */
+  private _destroyed = new Subject<void>();
+
+  /** The level of depth of the tree node. The padding will be `level * indent` pixels. */
+  @Input('cdkTreeNodePadding')
+  set level(value: number) {
+    this._level = value;
+    this._setPadding();
+  }
+  _level: number;
+
+  /** The indent for each level. Default number 40px from material design menu sub-menu spec. */
+  @Input('cdkTreeNodePaddingIndent')
+  set indent(value: number) {
+    this._indent = value;
+    this._setPadding();
+  }
+  _indent: number = 40;
```

Comment:
Should this support people setting a string with units? E.g. `10em`. 

(could be in a follow-up PR, but maybe add a TODO)